### PR TITLE
[MISC] Update deps and switch to reusable workflows

### PR DESCRIPTION
**Pull request by dragomirp**
Update all python dependencies, use reusable workflows for integration tests

### CODE REVIEW

File: .github/renovate.json5
Comment (dragomirp) [inline .github/renovate.json5:1]: Mostly a copy paste from the postgresql operator

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -22,53 +22,26 @@ on:
...
-          - backend-integration
-          - client-relation-integration
-          - legacy-client-relation-integration
-          - legacy-client-relation-integration-admin
-          - bundle-integration
-#          - read-distribution-integration
-          - tls-integration
-    name: ${{ matrix.tox-environments }}
+        juju:
+          - agent: 3.1.8  # renovate: latest juju 3
```
Comment (dragomirp) [inline .github/workflows/ci.yaml:32]: `self-signed-certificates` only works on Juju 3

File: .github/workflows/release.yaml
Comment (dragomirp) [inline .github/workflows/release.yaml:1]: Shellcheck changes

**Review by taurus-forever — APPROVED** [review]
Thank you!

**Pull request merged**